ports/stm32/accel.c

```diff
@@ -226,3 +232,180 @@ const mp_obj_type_t pyb_accel_type = {
+
+
+STATIC void accel_start(void) {
+    // start the I2C bus in master mode
+    i2c_init(I2C1, MICROPY_HW_I2C1_SCL, MICROPY_HW_I2C1_SDA, 400000, I2C_TIMEOUT_MS);
+
+    // readout WHO_AM_I register to check KXTJ3 device presence
+    uint8_t data[2] = { KXT_REG_WHO_AM_I };
+    i2c_writeto(I2C1, KXT_ADDR, data, 1, false);
+    i2c_readfrom(I2C1, KXT_ADDR, data, 1, true);
+    if ( data[0] != 0x35 ) {
+        mp_raise_msg(&mp_type_OSError, "accelerometer not found");
+    }
+
+    // set operating mode (default: 8 bits)
+    data[0] = KXT_REG_CTRL_REG1;
+    data[1] = 0x80;
+    i2c_writeto(I2C1, KXT_ADDR, data, 2, true);
+
+}
+
+/******************************************************************************/
+/* MicroPython bindings                                                      */
+
+#define NUM_AXIS (3)
+#define FILT_DEPTH (4)
+
+typedef struct _pyb_accel_obj_t {
+    mp_obj_base_t base;
+    int16_t buf[NUM_AXIS * FILT_DEPTH];
+} pyb_accel_obj_t;
+
+STATIC pyb_accel_obj_t pyb_accel_obj;
+
+/// \classmethod \constructor()
+/// Create and return an accelerometer object.
+///
+/// Note: if you read accelerometer values immediately after creating this object
+/// you will get 0.  It takes around 20ms for the first sample to be ready, so,
+/// unless you have some other code between creating this object and reading its
+/// values, you should put a `pyb.delay(20)` after creating it.  For example:
+///
+///     accel = pyb.Accel()
+///     pyb.delay(20)
+///     print(accel.x())
+STATIC mp_obj_t pyb_accel_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
+    // check arguments
+    mp_arg_check_num(n_args, n_kw, 0, 0, false);
+
+    // init accel object
+    pyb_accel_obj.base.type = &pyb_accel_type;
+    accel_start();
+
+    return MP_OBJ_FROM_PTR(&pyb_accel_obj);
+}
+
+STATIC mp_obj_t read_axis(int axis) {
+    uint8_t data[1] = { axis };
+    i2c_writeto(I2C1, KXT_ADDR, data, 1, false);
+    i2c_readfrom(I2C1, KXT_ADDR, data, 1, true);
+    return mp_obj_new_int(KXT_AXIS_SIGNED_VALUE(data[0]));
+}
+
+/// \method x()
+/// Get the x-axis value.
+STATIC mp_obj_t pyb_accel_x(mp_obj_t self_in) {
+    return read_axis(KXT_REG_XOUT_H);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(pyb_accel_x_obj, pyb_accel_x);
+
+/// \method y()
+/// Get the y-axis value.
+STATIC mp_obj_t pyb_accel_y(mp_obj_t self_in) {
+    return read_axis(KXT_REG_YOUT_H);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(pyb_accel_y_obj, pyb_accel_y);
+
+/// \method z()
+/// Get the z-axis value.
+STATIC mp_obj_t pyb_accel_z(mp_obj_t self_in) {
+    return read_axis(KXT_REG_ZOUT_H);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(pyb_accel_z_obj, pyb_accel_z);
+
+/// \method tilt()
+/// Not implemented with KXTJ3 accelerometer
+
+/// \method filtered_xyz()
+/// Not implemented with KXTJ3 accelerometer
+
+/// \method xyz()
+/// Get a 3-tuple of x, y and z values.
+STATIC mp_obj_t pyb_accel_xyz(mp_obj_t self_in) {
```

Comment:
I would call this just `filtered_xyz`.  It doesn't matter if it's not filtered, at least it gives the same API as the MMA.

Reply:
Done, i did changes and keep some filtering with the .filtered_xyz()
Thanks for your advice.